src/components/contextual-menu/contextual-menu.tsx

```diff
@@ -0,0 +1,127 @@
+import { Component, Prop, h, Host, Element, Watch, State } from "@stencil/core";
+
+@Component({
+  tag: "gxg-contextual-menu",
+  styleUrl: "contextual-menu.scss",
+  shadow: true
+})
+export class ContextualMenu {
+  constructor() {
+    this.detectClickOutsideMenu = this.detectClickOutsideMenu.bind(this);
+  }
+
+  /**
+   * The presence of this attribute makes the menu visible
+   */
+  @Prop({ reflect: true }) visible = false;
+
+  private mouseCoordinates = {
+    x: 0,
+    y: 0
+  };
+
+  private contextualMenuSizes = {
+    width: 0,
+    height: 0
+  };
+
+  @State() firstRightClick = true;
+
+  @Element() el: HTMLElement;
+
+  @Watch("visible")
+  watchHandler(newValue: boolean) {
+    if (newValue) {
+      //Get contextualMenu height and width
+      const contextualMenuWidth = (this.contextualMenuSizes.width = this.el.offsetWidth);
+      const contextualMenuHeight = (this.contextualMenuSizes.height = this.el.offsetHeight);
```

Comment:
I usually avoid as much as possible making assignments inside an expression, as many times the assignment is left "hidden" and hard to spot, which makes the code hard to understand and maintain. I recommend in these cases splitting it in two sentences:

```suggestion
      this.contextualMenuSizes.width = this.el.offsetWidth;
      const contextualMenuWidth = this.contextualMenuSizes.width;
      this.contextualMenuSizes.height = this.el.offsetHeight;
      const contextualMenuHeight = this.contextualMenuSizes.height;
```